stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1619,6 +1619,76 @@ const NumberArray{T<:Number} = AbstractArray{T}
 exactdiv(a, b) = a/b
 exactdiv(a::Integer, b::Integer) = div(a, b)
 
+# Bareiss algorithm
+function bareiss_update!(zero!, M::StridedMatrix, k, swapto, pivot, prev_pivot)
+    for i in k+1:size(M, 2), j in k+1:size(M, 1)
+        M[j,i] = exactdiv(M[j,i]*pivot - M[j,k]*M[k,i], prev_pivot)
+    end
+    zero!(M, k+1:size(M, 1), k)
+end
+
+function bareiss_update!(zero!, M::AbstractMatrix, k, swapto, pivot, prev_pivot)
+    V = @view M[k+1:end, k+1:end]
+    V .= @views exactdiv.(V * pivot - M[k+1:end, k] * M[k, k+1:end]', prev_pivot)
```

Comment:
Seems cleaner to just put `@views` on the whole function than having it on two lines.

Reply:
I'm still confused about why you would want this version, which allocates a bunch of temporary matrices

(Wouldn't you at least want `V .* pivot .-  ...` to fuse the two operations.)

Reply:
good point. Operations fused.